pkg/converter/converter_test.go

```diff
@@ -0,0 +1,171 @@
+kind: IstioOperator
+metadata:
+  name: default
+spec:`,
+			converterArgs: fmt.Sprintf("%s %s -n istio-system -v v1.24.3", istioFile, sailFile),
+			expectedOutput: `apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: v1.24.3
+  namespace: istio-system`,
+		},
+		{
+			name: "complex",
+			input: `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:
+  components:
+    base:
+      enabled: true
+    pilot:
+      enabled: false
+  values:
+    global:
+      externalIstiod: true
+      operatorManageWebhooks: true
+      configValidation: false
+    base:
+      enableCRDTemplates: true
+    pilot:
+      env:
+        PILOT_ENABLE_STATUS: true`,
+			converterArgs: fmt.Sprintf("%s %s -v v1.24.3 -n istio-system", istioFile, sailFile),
+			expectedOutput: `apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  values:
+    global:
+      externalIstiod: true
+      operatorManageWebhooks: true
+      configValidation: false
+    base:
+      enableCRDTemplates: true
+      enabled: true
+    pilot:
+      env:
+        PILOT_ENABLE_STATUS: "true"
+      enabled: false
+  namespace: istio-system
+  version: v1.24.3`,
+		},
+		{
+			name: "mandatory-arguments-only",
+			input: `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:`,
+			converterArgs: istioFile,
+			expectedOutput: `apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  namespace: istio-system`,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Cleanup(func() {
+				os.Remove(istioFile)
+				os.Remove(sailFile)
+			})
+			// Write the Istio YAML to the file
+			err := os.WriteFile(istioFile, []byte(tc.input), 0o644)
+			g.Expect(err).To(Succeed(), "failed to write YAML file")
+
+			// Run the configuration converter
+			err = executeConfigConverter(tc.converterArgs)
+			g.Expect(err).To(Succeed(), "error in execution of ./configuration-converter.sh")
+
+			// Read the output from the converted sail file
+			actualOutput, err := os.ReadFile(sailFile)
+			g.Expect(err).To(Succeed(), fmt.Sprintf("Cannot read %s", sailFile))
+
+			// Parse the actual output YAML
+			actualData, err := parseYaml(actualOutput)
+			g.Expect(err).To(Succeed(), "Failed to parse sailFile")
+
+			// Parse the expected output YAML
+			expectedData, err := parseYaml([]byte(tc.expectedOutput))
+			g.Expect(err).To(Succeed(), "Failed to parse expected output")
+
+			// Compare the actual and expected output
+			diff := cmp.Diff(actualData, expectedData)
+			g.Expect(diff).To(Equal(""), "Conversion is not as expected")
+		})
+	}
+}
+
+func executeConfigConverter(scriptArgs string) error {
+	converter := filepath.Join(project.RootDir, "tools", "configuration-converter.sh")
+
+	cmdArgs := []string{converter, scriptArgs}
+
+	cmd := strings.Join(cmdArgs, " ")
+	_, err := shell.ExecuteCommand(cmd)
```

Comment:
Can be simplified to:
```suggestion
	_, err := shell.ExecuteCommand(converter + " " + scriptArgs)
```